internal/controller/managedcluster_controller.go

```diff
@@ -683,6 +695,357 @@ func (r *ManagedClusterReconciler) machinesAvailable(ctx context.Context, namesp
 	return len(itemsList.Items) != 0, nil
 }
 
+func (r *ManagedClusterReconciler) reconcileCredentialPropagation(ctx context.Context, l logr.Logger, managedCluster *hmc.ManagedCluster) (ctrl.Result, error) {
+	l.Info("Reconciling CCM credentials propagation")
+	var err error
+	defer func() {
```

Comment:
it's deffered already, you don't need another one

Reply:
I had an issue with conditions not populating properly without it in this scope. I will recheck

Reply:
Rechecked and it looks I was wrong or missed something. So this is indeed extra, removed.